src/Microsoft.OData.Core/UriParser/Parsers/SegmentArgumentParser.cs

```diff
@@ -249,6 +249,14 @@ private bool TryConvertValue(IEdmTypeReference typeReference, string valueText,
                 return false;
             }
 
+            if (valueText[0] == '@')

```

Comment:
@mikepizzo Could `valueText` ever be an empty string? If so, this might result in an exception being thrown. Would it be worth adding a safeguard or clarification?

Reply:
Good call. It will never be null in our implementation, but the method that calls it is virtual so in theory someone could override the method and pass in a null value. I've added the suggested check.